roles/update_etc_hosts/tasks/update_host_present.yml

```diff
@@ -0,0 +1,149 @@
+---
+# TODO CHECK/SET Variables
+- name: Verify that variable node_ip is set
+  ansible.builtin.assert:
+    that: not( ( thishost.node_ip is undefined) or ( thishost.node_ip is none) or ( thishost.node_ip | trim == '') )
+    msg: "Variable 'node_ip' is undefined or empty. Please define it your host list"
+
+- name: Verify that variable node_ip is in the correct format
+  ansible.builtin.assert:
+    that: thishost.node_ip is ansible.utils.ip
+    msg: "Variable 'node_ip' is not an IP adress. Please use the correct format"
+
+- name: Verify that variable node_name is set
+  ansible.builtin.assert:
+    that: not( ( thishost.node_name is undefined) or ( thishost.node_name is none) or ( thishost.node_name | trim == '') )
+    msg: "Variable 'node_name' is undefined or empty. Please define it your host list"
+
+- name: Ensure node_domain is set
+  ansible.builtin.set_fact:
+    __update_etc_hosts_domain: "{{ thishost.node_domain | default(sap_domain | default(ansible_domain)) }}"
```

Comment:
Change 

"{{ thishost.node_domain | default(sap_domain | default(ansible_domain)) }}" -> 
"{{ thishost.node_domain | default(sap_domain) | default(ansible_domain) }}"

might improve readability, initially I thought you missed closing parenthesis

Reply:
will change, didn't know this works that way.